comp/metadata/packagesigning/packagesigningimpl/packagesigning.go

```diff
@@ -0,0 +1,136 @@
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"go.uber.org/fx"
+)
+
+// Module defines the fx options for this component.
+var Module = fxutil.Component(
+	fx.Provide(newpackagesigningProvider),
+)
+
+const defaultCollectInterval = 12 * time.Hour
+
+// Payload handles the JSON unmarshalling of the metadata payload
+type Payload struct {
+	Hostname  string           `json:"hostname"`
+	Timestamp int64            `json:"timestamp"`
+	Metadata  *signingMetadata `json:"signing_metadata"`
+}
+
+// MarshalJSON serialization a Payload to JSON
+func (p *Payload) MarshalJSON() ([]byte, error) {
+	type PayloadAlias Payload
+	return json.Marshal((*PayloadAlias)(p))
+}
+
+// SplitPayload implements marshaler.AbstractMarshaler#SplitPayload.
+// TODO implement the split
+func (p *Payload) SplitPayload(_ int) ([]marshaler.AbstractMarshaler, error) {
+	return nil, fmt.Errorf("could not split inventories host payload any more, payload is too big for intake")
+}
+
+type signingMetadata struct {
+	SigningKeys []SigningKey `json:"signing_keys"`
+}
+
+type pkgSigning struct {
+	util.InventoryPayload
+
+	log      log.Component
+	conf     config.Component
+	data     *signingMetadata
+	hostname string
+}
+
+type dependencies struct {
+	fx.In
+
+	Log        log.Component
+	Config     config.Component
+	Serializer serializer.MetricSerializer
+}
+
+type provides struct {
+	fx.Out
+
+	Comp          packagesigning.Component
+	Provider      runnerimpl.Provider
+	FlareProvider flaretypes.Provider
+}
+
+func newpackagesigningProvider(deps dependencies) provides {
+	hname, _ := hostname.Get(context.Background())
+	is := &pkgSigning{
+		conf:     deps.Config,
+		log:      deps.Log,
+		hostname: hname,
+		data:     &signingMetadata{},
+	}
+	is.InventoryPayload.SetIntervals(defaultCollectInterval, defaultCollectInterval)
+	is.InventoryPayload = util.CreateInventoryPayload(deps.Config, deps.Log, deps.Serializer, is.getPayload, "signing.json")
+
+	return provides{
+		Comp:          is,
+		Provider:      is.MetadataProvider(),
+		FlareProvider: is.FlareProvider(),
+	}
+}
+
+func (is *pkgSigning) fillData() {
+	if runtime.GOOS == "linux" {
+		pkgManager := getPackageManager()
+		switch pkgManager {
+		case "apt":
+			is.data.SigningKeys = getAPTSignatureKeys()
+		default: // yum and zypper are similar
+			is.data.SigningKeys = getYUMSignatureKeys(pkgManager)
+		}
+	}
+}
+
+func (is *pkgSigning) getPayload() marshaler.JSONMarshaler {
+	is.fillData()
+
+	return &Payload{
+		Hostname:  is.hostname,
+		Timestamp: time.Now().UnixNano(),
+		Metadata:  is.data,
+	}
+}
+
+// GetLinuxPackageSigningPolicy returns the global GPG signing policy of the host
+func GetLinuxPackageSigningPolicy() bool {
+	if runtime.GOOS == "linux" {
+		pkgManager := getPackageManager()
+		switch pkgManager {
+		case "apt":
+			return getNoDebsig()
+		default: // yum and zypper are similar
+			return getMainGPGCheck(pkgManager)
```

Comment:
I believe that for yum/dnf/zypper we should also collect the global value of repo_gpgcheck.